### PR TITLE
Add `dropout_node` to `torch_geometric.utils`

### PR DESCRIPTION
**Pull request by EdisonLeeeee**
This PR is my first attempt to add graph augmentation methods into PyG. As discussed in https://github.com/pyg-team/pytorch_geometric/issues/5452, I start with a simple augmentation method `DropNode` proposed in [Graph Contrastive Learning with Augmentations](https://arxiv.org/abs/2010.139023).

### CODE REVIEW

**Comment by EdisonLeeeee** [comment]
Planning to add these methods in the following PRs:
+ DropPath: [askGAE: Masked Graph Modeling Meets Graph Autoencoders](https://arxiv.org/abs/2205.10053)
+ FastGCN sampling: [FastGCN: Fast Learning with Graph Convolutional Networks via Importance Sampling](https://arxiv.org/abs/1801.10247)
+ LADIES: [Layer-Dependent Importance Sampling for Training Deep and Large Graph Convolutional Networks](https://arxiv.org/abs/1911.07323)
+ Add edges based on random walks: [Pairwise Learning for Neural Link Prediction](https://arxiv.org/pdf/2112.02936.pdf)
+ Node Shuffling
+ MaskFeature

File: torch_geometric/utils/dropout.py [hunk 1]
```diff
@@ -79,3 +82,45 @@ def dropout_adj(
...
+
+    if not training or p == 0.0:
+        return edge_index, edge_attr
+
+    num_nodes = maybe_num_nodes(edge_index, num_nodes)
+    nodes = torch.arange(num_nodes, dtype=torch.long, device=edge_index.device)
+    mask = torch.full_like(nodes, 1 - p, dtype=torch.float32)
+    mask = torch.bernoulli(mask).to(torch.bool)
+    subset = nodes[mask]
+    return subgraph(subset, edge_index, edge_attr, num_nodes=num_nodes)
```
Comment (rusty1s) [inline torch_geometric/utils/dropout.py:126]:
```suggestion
    return subgraph(mask, edge_index, edge_attr, num_nodes=num_nodes)
```
`subgraph` should already support `mask`.

**Review by rusty1s — COMMENTED** [review]
Thanks for adding this. Can we also create a small test for this?

**Comment by EdisonLeeeee** [comment]
Of course. Will add some tests after addressing your comments :)

File: torch_geometric/utils/dropout.py [hunk 2]
```diff
@@ -79,3 +82,45 @@ def dropout_adj(
         edge_index = torch.stack([row, col], dim=0)
 
     return edge_index, edge_attr
+
+
+def dropout_node(edge_index: Tensor, edge_attr: OptTensor = None,
```
[the remark this answers is not in the document]
Comment (EdisonLeeeee) [reply torch_geometric/utils/dropout.py]: Sure. A quick question, shall we also return a tuple of `edge_index` and `edge_mask` for `dropout_adj` to align them? I think returning `edge_mask` would be more flexible for most cases.

[the remark this answers is not in the document]
Comment (rusty1s) [reply torch_geometric/utils/dropout.py]: Yeah, I think we should align the interfaces. At this point, it is probably better to just introduce a new function called `dropout_edge` and deprecate the old one.

[the remark this answers is not in the document]
Comment (EdisonLeeeee) [reply torch_geometric/utils/dropout.py]: Got it. One more question, shall we add support for `SparseTensor`. If it is necessary, I can update the code in the next commit.

[the remark this answers is not in the document]
Comment (rusty1s) [reply torch_geometric/utils/dropout.py]: I don't think this is necessary for now. Let's leave it as it is.

File: torch_geometric/utils/dropout.py [hunk 3]
```diff
@@ -79,3 +82,56 @@ def dropout_adj(
...
+def dropout_node(edge_index: Tensor, p: float = 0.5,
+                 num_nodes: Optional[int] = None,
+                 training: bool = True) -> Tuple[Tensor, Tensor, Tensor]:
+    r"""Randomly drops nodes from the adjacency matrix
+    :obj:`edge_index` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`edge_index`, (2) the edge mask
+    indicating which edges were dropped. (3) the node mask indicating
+    which nodes were dropped.
```
Comment (rusty1s) [inline torch_geometric/utils/dropout.py:96]: nit: I would slightly prefer `(edge_index, node_mask, edge_mask)` :D

**Review by rusty1s — APPROVED** [review]
Thank you!

File: torch_geometric/utils/dropout.py (hunk 3 above)
Comment (rusty1s) [reply torch_geometric/utils/dropout.py:96]: Ok, makes sense :)

**Comment by rusty1s** [comment]
@EdisonLeeeee Can you also add the changes to `CHANGELOG.md`?

**Comment by EdisonLeeeee** [comment]
@rusty1s done. Will add `dropout_edge` and deprecate `dropout_adj` in the next PR.
